run-clang-format.py

```diff
@@ -0,0 +1,408 @@
+#!/usr/bin/env python
```

Comment:
**suggestion:** The wrapper script adds some really conveniences on top of the default `clang-format` command.

- Parallel execution. The sequential run is still pretty fast given the small codebase.
- Code diff output.

On the other hand, this adds a new dependency to the project, that is `python3`. While most modern OS comes bundled with Python 3, Windows might require additional setup.

Weighting between cost vs benefits, I wonder if there is the right time to introduce the wrapper script 🤔. I could have missed the rationale behind this change though. Keen to hear your thoughts.

Reply:
I was actually thinking the same thing, linux ( maybe mac, have no experience with mac so not sure) has python3 as default package, and to install python on windows is pretty straight forward. I did consider the windows users who have not installed python. That's why I use the ` find . regex .... ` command in the makefile and Readme. But I think we can keep the python script because I think it easlier to use than ` clang-format ` command(e.g. simply use option ` -r ` can go through every files and color indication). We can have both command (script and `clang-format`) introduce in the readme file. Therefore, its not mandatory for windows users install python.